### PR TITLE
Search for an invalid postcode

### PR DESCRIPTION
**Pull request by stevenwilkin**
Any opinions on the use of exceptions?

### CODE REVIEW

**Comment by gjvis** [comment]
Maybe a bit dangerous to use exceptions for control flow (it's not really an exceptional case), but the code is easy to follow. Your call

File: app/controllers/locations_controller.rb [hunk 1]
```diff
@@ -7,6 +7,9 @@ def index
     @locations = locations.map do |location|
       SearchResultDecorator.new(location)
     end
+
+  rescue Geocoder::InvalidPostcode
+    render :invalid_postcode
```
Comment (andrewgarner) [inline app/controllers/locations_controller.rb]: Wouldn't we want to show the user an error message not an error page?

**Comment by andrewgarner** [comment]
Agree on flow. If we take the responsibility of postcode validation away from `Geocoder` we don't need to handle exceptions in the way we do. We could just invoke `UKPostcode` directly from the controller or if we think the controller is doing too much wrap those invocations up as another object.

**Comment by stevenwilkin** [comment]
My initial thought was to do the postcode validation in the controller but in the scenario where the user supplies an **invalid postcode** which is actually a **valid outcode** we're planning on looking up the postcode and geocoding that.

As discussed with @gjvis, we're going to ice this PR until we know what the UI is going to look like which will tell us what the API for this code should look like. This should solve the error message versus error page problem as well.

**Pull request merged**